Apps/FlightSoftware/Watchdog/src/main.c

```diff
@@ -166,7 +171,7 @@ int main(void) {
 
         /* a cool thing happened! now time to check what it was */
         if (loop_flags & FLAG_UART0_RX_PACKET) {
-            watchdog_handle_hercules(); // @suppress("Invalid arguments")
+            watchdog_handle_hercules(i2cReadings); // @suppress("Invalid arguments")

```

Comment:
Does this compile? For one, I don't think the signature of `watchdog_handle_hercules` accepts an argument. Also, here you are using a direct copy pass, whereas in `watchdog_handle_hercules` you are accepting a pointer.

Reply:
Ah it only compiled because of that `// @suppress("Invalid arguments")` , I will fix this accordingly whether you prefer the global i2cReadings route or the passing-as-an-argument i2cReadings route

Reply:
Yeah, I'm not a fan of comments suppressing warnings without an accompanying comment explaining why it is appropriate to do so in this particular case. There's a reason we compile with `-Wall -Werror`, and suppressing warnings just goes around that. If we can I'd prefer to remove the suppression.  In C you should pretty much always pass structs by pointer so that you're not copying the entire struct (and in C++ you should usually pass structs and class by reference, for the same reason).